sam/bamWrite.go

```diff
@@ -275,6 +276,10 @@ func writeExtra(bw *BamWriter, s Sam) {
 // retrieveTriplet splits a tag string into type, number, and value.
 func retrieveTriplet(tag string) []string {
 	comp := strings.Split(tag, ":")
+	if comp[1] == "B" {
```

Comment:
1) would this function effectually run strictly if `len(comp) = 3 or greater than 3`?
2) Do we only return exactly 3 items due to `comp[:3]`
I tried something along the lines of this and it cleans up the need to check the len multiple times. It passed your tests, but I didn't do a deeper dive into the functions as the bam writer is pretty complex:

```golang
func retrieveTriplet(tag string) []string {
	comp := strings.Split(tag, ":")
	if len(comp) >= 3 {
		if comp[1] == "B" {
			comp[1] = comp[1] + comp[2][0:1] // the B case has another value as part of the array that needs to be stored in the second part of the triplet
			comp[2] = comp[2][2:]  
		}
	} else {
		log.Panicf("malformed auxiliary data '%s'", tag)
	}
	comp[2] = strings.Join(comp[2:], ":") // rejoin value component of tag
	comp = comp[:3]
	return comp
}
```

Reply:
Yes, the return should always be len of 3, though comp may be >= 3. 

I incorporated your suggestion, though with minor edits to keep the check that the members of the triplet are the correct number of characters.